### DISCUSSION TITLE
Option Picker working well on an app but not for the other

### DISCUSSION THREAD

**Discussion by LMPexplorer** [Budibase/budibase#13790] [Help] [answered]
**Checklist**
- [ ] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**

- Self
  - Budibase Version: <version>2.27.3. 
  - App Version: <version> 2.27.3. 

**Describe the bug**
Hi, I am running latest Version and using PostgreSQL DB.
I am currently trying to create an option picker that will show some texts (from a text column) and filter on pick for a Data Provider.
My issue here is that i have 2 different apps with similar database :
- The first one works really well, I can pick one of my texts well and it will filter as i want
- In the second app, by doing the exact same steps, it doesn't work, only one of my texts is displayed on pick. 
I tried to understand the issue but after trying to recreate it multiple times i can't seem to find the issue.
I will show some screenshots from both app to help understand the issue.

**To Reproduce**
1. Go to POSTGRESQL DB, create a table with one text colum with multiples rows with differents text values
2. Go to design, create a Data Provider with the table you created, nest a form (with same table and update setting) and an option picker
3. In Settings of the option picker, chose Option source : DataProvider, chose you data provider, and label and value the name of your column with the differents texts
4. See that after this, when you clic on the option picker, there will only be the first text displayed but not the others

**Expected behavior**
As it works well in my first app, i would like to be able to choose any of the text present in my column

**Screenshots**
**_First app working WELL :_**
Component tree :
![image](https://github.com/Budibase/budibase/assets/158468320/885d249d-ba87-4635-927e-a97c86364a44)
Option picker settings (Nom_du_projet is my column with differents texts) :
![image](https://github.com/Budibase/budibase/assets/158468320/c391f490-f999-423d-a043-0d9008653148)
View of the differents text displayed :
![image](https://github.com/Budibase/budibase/assets/158468320/4f9f2bf9-7db6-489c-9504-0984d4cd3aa3)
Screen of the column in database : 
![image](https://github.com/Budibase/budibase/assets/158468320/c2dea219-0d67-47b5-ac78-6d27e123d541)

**_ Second app NOT WORKING :_**
Component tree :
![image](https://github.com/Budibase/budibase/assets/158468320/6a4c4a38-5944-467a-82dd-bf5d44b01852)
Option picker settings (nom_simu = Nom_du_projet in the other app) :
![image](https://github.com/Budibase/budibase/assets/158468320/b3dd4315-fa74-4719-aaa9-a7384fc3f985)
View of the differents text displayed :
![image](https://github.com/Budibase/budibase/assets/158468320/c0174435-32c3-4dd9-a4ee-f73a80f47b41)
Screen of the column in database : 
![image](https://github.com/Budibase/budibase/assets/158468320/b9339c68-cde7-4778-97a9-ef08c61f5e65)

  
**App Export**
//

Thanks

**Comment by linear** [comment]
<p><a href="https://linear.app/budibase/issue/BUDI-8299/option-picker-working-well-on-an-app-but-not-for-the-other">BUDI-8299 Option Picker working well on an app but not for the other</a></p>

**Comment by aptkingston** [comment, marked as answer]
Hi @LMPexplorer. If you're using a data provider as the source of the options picker, it will only show values that are present in the rows it can see. In this case, you just need to load more rows. What's happening is that the default amount of rows (50) all contain the same value (`test`) so it is the only value it can see.

If you increase the amount of rows loaded by the data provider so that it includes all rows in your table then this will work as expected. e.g. try increasing it to 100 or 200 rows (or as many as you need for the full table).

**Reply by LMPexplorer** [reply]
Of course it is the limitation in the data provider !!! I feel dumb sometimes ahah thank you very much for your quick response, you can put the subject as answered.
Have a nice day.
